File: Frequencies of Limited Range Array Elements/solution1.java
import java.util.*;

class Solution{
    public static void frequencyCount(int arr[], int N, int P)
    {
        HashMap<Integer, Integer> map = new HashMap<>();
        for(int i=0;i<N;i++) {
            if(map.containsKey(arr[i])) {
                map.put(arr[i], map.get(arr[i]) + 1);
            } else {
                map.put(arr[i], 1);
            }
        }
        // for(Integer key: map.keySet()) System.out.println("key -> " + key + " " + "value -> " + map.get(key));
        for(int i=0;i<N;i++) {
            if(map.containsKey(i+1)) {
                System.out.println("arr[i] -> " + arr[i] +  " map get -> " + map.get(i+1) + " i -> " + i);
                arr[i] = map.get(i+1);
            } else {
                // System.out.println("value else -> " + arr[i]);
                arr[i] = 0;
            }
        }
    }
}

public class solution1 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int[] arr = new int[N];
        for(int i=0;i<N;i++) arr[i] = sc.nextInt();
        int P = sc.nextInt();
        Solution.frequencyCount(arr, N, P);
        for(int i: arr) System.out.print(arr[i] + " ");
    }
}

File: Frequencies of Limited Range Array Elements/solution1.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution{
    public:
    void frequencyCount(vector<int>& arr,int N, int P) { 
        map<int, int> m;
        for(auto i: arr) m[i]++;
        for(int i=0;i<arr.size();i++) {
            if(m.find(i+1) != m.end()) arr[i] = m[i+1]; 
            else arr[i] = 0;
        }
    }
};
int main() {
    int n;
    cin>>n;
    vector<int> arr(n);
    for(int i=0;i<n;i++) cin>>arr[i];
    int p; cin>>p;
    Solution sol;
    sol.frequencyCount(arr,n,p);
    for(auto i: arr) cout<<i<<" ";
}